Admixed strains identified via admixture were removed from the vcf by using vcftools 

```
vcftools --vcf ../02.Admixture/Suvarum.recode.vcf --remove admixed.txt --recode-INFO-all --non-ref-ac-any 1 --out Suva_NonAdmixed --recode
```
Then, the nonAdmixed vcf was prunned same as the admixture analysis

```
bash /plink_pruning_prep.sh nonAdmixed.vcf
```
then, using the output from that script we run:

```
plink --allow-extra-chr --double-id --indep-pairwise 50 5 0.5 --make-bed --out ld_prunned --vcf nonAdmixed.recode_annot.vcf
```
Change “_” for tabulation 
```
sed 's/_/\t/' ld_prunned.prune.in > ld_prunned.prune.in.vcftools
```
we extract the marked positions from the nonAdmixed.vcf file by using the ld_prunned.prune.in.vcftools file.
```
vcftools --vcf nonAdmixed.vcf --recode-INFO-all --recode --out Suva_LDfilter --positions ld_prunned.prune.in.vcftools
```
The Suva_LDfilter.vcf file is transformed to plink format
```
plink --allow-extra-chr --double-id --make-bed --out plinkpca_ldprunned  --vcf Suva_LDfilter.vcf
```
Then, the plink format is transformed to eigenstrat format
```
./fam2ind.sh plinkpca_ldprunned.fam # made by courtesy of Dr A. Altmann
./bim2snp.sh plinkpca_ldprunned.bim # made by courtesy of Dr A. Altmann
./bed2geno.sh plinkpca_ldprunned.bed (modified to allow non canonical chromosome names --allow-extra-chr) # made by courtesy of Dr A. Altmann
extracted from https://rpubs.com/maffleur/452627
```
In case of need chromosome names changes
```
sed -i '' 's/ChrXVI/16/g' Suva.LD.positions.bim
```
Parameters file was cosntructed with ind, snp and geno files, named smartpca.pam
```
genotypename: plinkpca_ldprunned.geno
snpname: plinkpca_ldprunned.snp
indivname: plinkpca_ldprunned.ind
evecoutname: Suva.pca.evec
evaloutname: Suva.pca.eval
numoutevec: 10
numoutlieriter: 0
```
Finally, we run 
```
smartpca -p smartpca.pam
```
Plotting was made with PCA.R
